dynamic-tools/zap-extensions/wso2-jira-issue-creator/wso2jiraissuecreator/ExtensionJiraIssueCreater.java

```diff
@@ -0,0 +1,268 @@
+                                log.info("it is executing here");
+                                create_issues.setTitle("Create Jira Issues");
+                                create_issues.listJiraProjects();
+                                create_issues.show();
+                            }
+                            input.close();
+
+                        } catch (FileNotFoundException e) {
+                            log.error(e.getMessage(), e);
+                            View.getSingleton().showWarningDialog("Credential file not found !!");
+                            credFrm.show();
+
+                        } catch (IOException e) {
+                            log.error(e.getMessage(), e);
+
+                        } catch (AuthenticationException e) { //jira throws a capcha user has to log and try again
+                            cred_file.delete();
+                            log.error(e.getMessage(), e);
+                            View.getSingleton().showWarningDialog(
+                                    "Wrong Credentials! Please login to your jira account and retry!!");
+
+                        }
+
+                    } else { //create credential file if not found
+
+                        credFrm.show();
+
+                    }
+
+                }
+            });
+        }
+        return menuExample;
+    }
+
+    @Override public String getAuthor() {
+        return Constant.messages.getString(IssueCreatorConstants.PREFIX + ".author");
+    }
+
+    @Override public String getDescription() {
+        return Constant.messages.getString(IssueCreatorConstants.PREFIX + ".desc");
+    }
+
+    @Override public URL getURL() {
+        try {
+            return new URL(Constant.ZAP_EXTENSIONS_PAGE);
+        } catch (MalformedURLException e) {
+            return null;
+        }
+    }
+
+    /**
+     * api methods
+     **/
+
+    private static boolean stringToBool(String s) {
+        if (s.equals("1"))
+            return true;
+        if (s.equals("0"))
+            return false;
+        throw new IllegalArgumentException(s + " is not a bool. Only 1 and 0 are.");
+    }
+
+    public void createJiraIssues(String jiraBaseURL, String jiraUserName, String jiraPassword, String projectKey,
+            String asssignee, String high, String medium, String low,String issueLabel, String filterIssuesByResourceType) {
+
+        String project_key = projectKey;
+        String issueList[];
+        JiraRestClient jira = new JiraRestClient();
+        int issueCount;
+        String issue;
+
+        try {
+
+            String auth = new String(Base64.encode(jiraUserName + ":" + jiraPassword));
+            String BASE_URL = jiraBaseURL;
+
+            XmlDomParser xmlParser = new XmlDomParser();
+            if (high.equals("1") || medium.equals("1") || low.equals("1")) {
+                issueList = xmlParser.parseXmlDoc(project_key, asssignee, stringToBool(high), stringToBool(medium),
```

Comment:
Before parsing, check if the parser is properly configured to avoid XML attacks.